addons/web/static/src/search/favorite_menu/custom_favorite_item.js

```diff
@@ -3,16 +3,16 @@
 import { Dropdown } from "@web/core/dropdown/dropdown";
 import { CheckBox } from "@web/core/checkbox/checkbox";
 import { registry } from "@web/core/registry";
-import { useAutofocus, useService } from "@web/core/utils/hooks";
+import { useService } from "@web/core/utils/hooks";
 
-import { Component, useState } from "@odoo/owl";
+import { Component, useRef, useState } from "@odoo/owl";
 
 const favoriteMenuRegistry = registry.category("favoriteMenu");
 
 export class CustomFavoriteItem extends Component {
     setup() {
         this.notificationService = useService("notification");
-        this.descriptionRef = useAutofocus();
+        this.descriptionRef = useRef("autofocus");
```

Comment:
maybe rename the ref then?

Also, can this be easily enough tested?

Reply:
Hmm, I'll try to write a test. Could be easy.

Reply:
> maybe rename the ref then?
You're right. It should be named "autofocus".

Reply:
I'm trying this but doesn't work. Am I missing something so that manually triggered keyboard events be able to type the character in the input? If this is not really possible, can you suggest an alternative?

```js

            await toggleFavoriteMenu(target);
            await toggleSaveFavorite(target);
            await editFavoriteName(target, "My favorite 2");
            const input = focusFavoriteInput(target, 3); // Focus at the input and put the cursor at index 3

            // This block doesn't actually put the "*" character in the cursor position.
            const keydownEvent = new KeyboardEvent("keydown", { key: "*" });
            const keyupEvent = new KeyboardEvent("keyup", { key: "*" });
            input.dispatchEvent(keydownEvent);
            input.dispatchEvent(keyupEvent);
            await nextTick();

            // Therefore these checks don't work.
            assert.strictEqual(input.value, "My *favorite 2");
            assert.strictEqual(input.selectionEnd, 4);
```

And by the way, I also tried manually modifying the "input.value" by assigning a new value, but the cursor was moved at the end, both with this fix and without.